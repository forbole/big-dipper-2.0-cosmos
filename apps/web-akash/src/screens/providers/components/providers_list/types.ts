export type TitleListType = {
  ownerAddress: string;
  hostUri: string;
  region: string;
  organization: string;
  email: string;
  website: string;
};

export type TitleListState = {
  hasNextPage: boolean;
  isNextPageLoading: boolean;
  offsetCount: number;
  data: TitleListType[];
};
