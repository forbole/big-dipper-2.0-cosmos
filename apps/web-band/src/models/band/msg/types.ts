import type { BaseCategories } from 'ui/models/msg/types';
export type CustomCategories = 'oracle'; // custom modules
export type Categories = BaseCategories | CustomCategories;
