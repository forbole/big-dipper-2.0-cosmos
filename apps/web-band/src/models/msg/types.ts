import type { BaseCategories, Log } from 'ui/models/msg/types';

export type CustomCategories = 'oracle'; // custom modules
export type Categories = BaseCategories | CustomCategories;
export type { Log };
