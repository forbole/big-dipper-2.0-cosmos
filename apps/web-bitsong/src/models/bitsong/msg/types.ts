import type { BaseCategories } from 'ui/models/msg/types';

export type CustomCategories = 'nft' | 'auction' | 'fantoken'; // custom modules
export type Categories = BaseCategories | CustomCategories;
