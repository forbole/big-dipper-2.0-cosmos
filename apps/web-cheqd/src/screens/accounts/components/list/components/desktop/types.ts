import { CSSProperties, ReactNode } from 'react';
import { ItemType } from '@/screens/accounts/components/list/types';

export type DesktopProps = {
  className?: string;
  items: ItemType[] | undefined;
};

export type CellProps = {
  style: CSSProperties;
  rowIndex: number;
  children: ReactNode;
  cx: string;
};
export type HeaderProps = {
  style: CSSProperties;
  columnIndex: number;
};

export type RowProps = {
  data: ItemType;
  style: CSSProperties;
  rowIndex: number;
  columnIndex: number;
};
