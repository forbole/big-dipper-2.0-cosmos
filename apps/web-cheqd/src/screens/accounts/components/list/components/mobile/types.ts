import { ItemType } from '@/screens/accounts/components/list/types';

export type MobileProps = {
  className?: string | undefined;
  items: ItemType[] | undefined;
};

export type RowProps = {
  data: ItemType;
  index: number;
  itemCount: number | undefined;
};
