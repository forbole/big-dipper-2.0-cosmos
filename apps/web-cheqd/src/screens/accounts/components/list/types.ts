/**
 * `ItemType` is returned by the `useAccounts` hook.
 * @property {number} rank - The rank of the address in the top 100 addresses.
 * @property {string} address - The address of the account
 * @property {number} balance - The balance of the address in the token.
 * @property {number} percentage - The percentage of the total supply that the address holds.
 */
export type ItemType = {
  rank: number;
  address: string;
  balance: number;
  percentage: number;
};

/**
 * `UseAccountsState` is the return type of the `useAccounts` hook.
 * @property items - Array<ItemType>;
 * @property {boolean} loading - boolean - A boolean indicating whether the query is loading
 * or not.
 * @property {boolean} exists - boolean;
 * @property {number} page - The current page number.
 * @property setPage - Function to be called when the user changes the page.
 * @property {number} rowsPerPage - number;
 * @property setRowsPerPage - Function to be called when the user changes rowsPerPage;
 */
export type UseAccountsState = {
  items: Array<ItemType> | undefined;
  loading: boolean;
  exists: boolean;
  page: number;
  setPage: (page: number) => void;
  rowsPerPage: number;
  setRowsPerPage: (rowsPerPage: number) => void;
};
