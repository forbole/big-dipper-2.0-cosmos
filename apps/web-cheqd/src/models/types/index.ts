import type { BaseCategories, Log } from 'ui/models/msg/types';

export type CustomCategories = 'cheqd'; // custom modules
export type Categories = BaseCategories | CustomCategories;
export type { Log };
