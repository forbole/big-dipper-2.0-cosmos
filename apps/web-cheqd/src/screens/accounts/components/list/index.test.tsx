/* eslint-disable object-curly-newline */
import { TopAccountsDocument } from '@/graphql/types/general_types';
import List from '@/screens/accounts/components/list';
import MockTheme from '@/tests/mocks/MockTheme';
import { mockClient } from '@/tests/mocks/mockApollo';
import wait from '@/tests/utils/wait';
import { ApolloProvider } from '@apollo/client';
import { MockedProvider } from '@apollo/client/testing';
import renderer from 'react-test-renderer';

// ==================================
// mocks
// ==================================
const mockAccountBalancesDocument = jest.fn().mockReturnValue({
  data: {
    top_accounts: [
      {
        address: 'address1',
        available: 0,
        delegation: 0,
        redelegation: 0,
        reward: 0,
        sum: 0,
        unbonding: 0,
        type: '',
      },
      {
        address: 'address2',
        available: 0,
        delegation: 0,
        redelegation: 0,
        reward: 0,
        sum: 0,
        unbonding: 0,
        type: '',
      },
    ],
  },
});

// ==================================
// unit tests
// ==================================
describe('screen: Accounts/List', () => {
  it('matches snapshot', async () => {
    let component: renderer.ReactTestRenderer | undefined;

    renderer.act(() => {
      component = renderer.create(
        <ApolloProvider client={mockClient}>
          <MockedProvider
            mocks={[
              {
                request: {
                  query: TopAccountsDocument,
                  variables: {
                    limit: 100,
                    offset: 0,
                  },
                },
                result: mockAccountBalancesDocument,
              },
            ]}
          >
            <MockTheme>
              <List />
            </MockTheme>
          </MockedProvider>
        </ApolloProvider>
      );
    });
    await wait(renderer.act);

    const tree = component?.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
