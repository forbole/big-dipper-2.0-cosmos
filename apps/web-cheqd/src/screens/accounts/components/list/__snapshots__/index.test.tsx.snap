// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: Accounts/List matches snapshot 1`] = `
.emotion-0 {
  height: 100%;
}

.emotion-1 {
  min-height: 100vh;
}

.emotion-1.emotion-1 {
  overflow: auto;
  padding: 16px;
  border-radius: 4px;
  background: #131316;
}

.emotion-2 {
  padding: 1rem;
}

.emotion-3 {
  overflow: auto;
  color: #E6E6E6;
  font-size: 0.875rem;
}

.emotion-3:last-child {
  padding: 0;
}

.emotion-3 .MuiTablePagination-spacer {
  display: none;
}

.emotion-3 .MuiToolbar-gutters {
  padding: 0;
  margin: 0;
  -webkit-flex-direction: column-reverse;
  -ms-flex-direction: column-reverse;
  flex-direction: column-reverse;
  height: auto;
  min-height: initial;
}

.emotion-3 .MuiTablePagination-caption {
  -webkit-flex-shrink: initial;
  -ms-flex-negative: initial;
  flex-shrink: initial;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: flex-end;
  align-self: flex-end;
  margin-top: 16px;
  color: #818181;
}

@media (min-width:768px) {
  .emotion-3 .MuiToolbar-gutters {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-box-pack: end;
    -ms-flex-pack: end;
    -webkit-justify-content: flex-end;
    justify-content: flex-end;
  }

  .emotion-3 .MuiTablePagination-caption {
    margin-top: 0;
    margin-right: 16px;
  }
}

.emotion-3.emotion-3 {
  margin-top: 24px;
}

.emotion-4 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 16px;
  padding-right: 16px;
  min-height: 52px;
  padding-right: 2px;
}

@media (min-width:375px) {
  .emotion-4 {
    padding-left: 24px;
    padding-right: 24px;
  }
}

@media (min-width: 1280px) {
  .emotion-4 {
    height: 160px;
    overflow: hidden;
  }
}

@media (min-width:0px) and (orientation: landscape) {
  .emotion-4 {
    min-height: 52px;
  }
}

@media (min-width:375px) {
  .emotion-4 {
    min-height: 52px;
    padding-right: 2px;
  }
}

.emotion-4 .MuiTablePagination-actions {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-left: 20px;
}

.emotion-5 {
  -webkit-flex: 1 1 100%;
  -ms-flex: 1 1 100%;
  flex: 1 1 100%;
}

.emotion-6 {
  font-size: 0.875rem;
  letter-spacing: 0.25px;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  line-height: 1.43;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0;
  list-style-type: none;
}

.emotion-7 .MuiButtonBase-root,
.emotion-7 .MuiInputBase-input {
  background: #19191D;
}

.emotion-7 li {
  margin: 0px 4px;
}

.emotion-7 li.last {
  margin-right: 0;
  margin-left: 3px;
}

.emotion-7 li.first {
  margin-left: 0;
  margin-right: 3px;
}

@media (min-width:768px) {
  .emotion-7 {
    -webkit-align-self: flex-end;
    -ms-flex-item-align: flex-end;
    align-self: flex-end;
  }
}

@media (min-width:768px) {
  .emotion-7.emotion-7 {
    display: none;
  }
}

.emotion-8 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  padding: 8px;
  border-radius: 50%;
  overflow: visible;
  color: #fff;
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  padding: 12px;
  font-size: 1.75rem;
  height: 30px;
  width: 30px;
  border-radius: 4px;
  padding: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-8::-moz-focus-inner {
  border-style: none;
}

.emotion-8.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-8 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-8.Mui-disabled {
  background-color: transparent;
  color: rgba(255, 255, 255, 0.3);
}

.emotion-9 {
  -webkit-transform: rotate(180deg);
  -moz-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.emotion-12 {
  margin: 0;
  font-size: 0.875rem;
  letter-spacing: 0.25px;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  line-height: 1.43;
  height: 30px;
  width: 30px;
  border-radius: 4px;
  padding: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  color: #818181;
}

.emotion-12.selected {
  color: #FFFFFF;
  background: rgba(253,59,76,0.7);
}

.emotion-12.selected:hover {
  cursor: initial;
}

.emotion-12:hover {
  cursor: pointer;
}

.emotion-16 {
  display: none;
}

@media (min-width:768px) {
  .emotion-16 {
    display: inline-block;
  }
}

.emotion-17 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  min-width: 0;
  padding: 0;
  margin: 0;
  border: 0;
  vertical-align: top;
}

.emotion-18 {
  font-size: 1rem;
  white-space: pre-wrap;
  letter-spacing: 0.5px;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  line-height: 1.4375em;
  color: #E6E6E6;
  box-sizing: border-box;
  position: relative;
  cursor: text;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-18.Mui-disabled {
  color: rgba(255, 255, 255, 0.5);
  cursor: default;
}

.emotion-18 {
  border-radius: 12px;
}

.emotion-18 .MuiInputBase-input {
  padding: 0;
}

.emotion-18 .MuiTypography-body2 {
  color: #818181;
}

.emotion-18 .MuiSelect-selectMenu {
  height: 30px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border-radius: 4px;
  padding-left: 8px;
  padding-right: 24px;
}

.emotion-19 {
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border-radius: 4px;
  cursor: pointer;
  font: inherit;
  letter-spacing: inherit;
  color: currentColor;
  padding: 4px 0 5px;
  border: 0;
  box-sizing: content-box;
  background: none;
  height: 1.4375em;
  margin: 0;
  -webkit-tap-highlight-color: transparent;
  display: block;
  min-width: 0;
  width: 100%;
  -webkit-animation-name: mui-auto-fill-cancel;
  animation-name: mui-auto-fill-cancel;
  -webkit-animation-duration: 10ms;
  animation-duration: 10ms;
}

.emotion-19:focus {
  border-radius: 4px;
}

.emotion-19::-ms-expand {
  display: none;
}

.emotion-19.Mui-disabled {
  cursor: default;
}

.emotion-19[multiple] {
  height: auto;
}

.emotion-19:not([multiple]) option,
.emotion-19:not([multiple]) optgroup {
  background-color: #131316;
}

.emotion-19.emotion-19.emotion-19 {
  padding-right: 32px;
}

.emotion-19.MuiSelect-select {
  height: auto;
  min-height: 1.4375em;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

.emotion-19::-webkit-input-placeholder {
  color: currentColor;
  opacity: 0.5;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-19::-moz-placeholder {
  color: currentColor;
  opacity: 0.5;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-19:-ms-input-placeholder {
  color: currentColor;
  opacity: 0.5;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-19::-ms-input-placeholder {
  color: currentColor;
  opacity: 0.5;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-19:focus {
  outline: 0;
}

.emotion-19:invalid {
  box-shadow: none;
}

.emotion-19::-webkit-search-decoration {
  -webkit-appearance: none;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-19::-webkit-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-19::-moz-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-19:-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-19::-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-19:focus::-webkit-input-placeholder {
  opacity: 0.5;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-19:focus::-moz-placeholder {
  opacity: 0.5;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-19:focus:-ms-input-placeholder {
  opacity: 0.5;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-19:focus::-ms-input-placeholder {
  opacity: 0.5;
}

.emotion-19.Mui-disabled {
  opacity: 1;
  -webkit-text-fill-color: rgba(255, 255, 255, 0.5);
}

.emotion-19:-webkit-autofill {
  -webkit-animation-duration: 5000s;
  animation-duration: 5000s;
  -webkit-animation-name: mui-auto-fill;
  animation-name: mui-auto-fill;
}

.emotion-19 {
  padding: 8px 12px;
}

.emotion-20 {
  margin: 0;
  font-size: 0.875rem;
  letter-spacing: 0.25px;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  line-height: 1.43;
}

.emotion-21 {
  bottom: 0;
  left: 0;
  position: absolute;
  opacity: 0;
  pointer-events: none;
  width: 100%;
  box-sizing: border-box;
}

.emotion-22 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  position: absolute;
  right: 7px;
  top: calc(50% - .5em);
  pointer-events: none;
  color: #fff;
}

.emotion-22.Mui-disabled {
  color: rgba(255, 255, 255, 0.3);
}

.emotion-23 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0;
  list-style-type: none;
}

.emotion-23 .MuiButtonBase-root,
.emotion-23 .MuiInputBase-input {
  background: #19191D;
}

.emotion-23 li {
  margin: 0px 4px;
}

.emotion-23 li.last {
  margin-right: 0;
  margin-left: 3px;
}

.emotion-23 li.first {
  margin-left: 0;
  margin-right: 3px;
}

@media (min-width:768px) {
  .emotion-23 {
    -webkit-align-self: flex-end;
    -ms-flex-item-align: flex-end;
    align-self: flex-end;
  }
}

.emotion-23.emotion-23 {
  display: none;
}

@media (min-width:768px) {
  .emotion-23.emotion-23 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

<div
  className="emotion-0"
>
  <div
    className="emotion-1"
  >
    <div
      className="emotion-2"
    >
      accounts:refresh_delay_notice
    </div>
    <div
      className="MuiTablePagination-root emotion-3"
    >
      <div
        className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar emotion-4"
      >
        <div
          className="MuiTablePagination-spacer emotion-5"
        />
        <p
          className="MuiTablePagination-displayedRows emotion-6"
        >
          paginationLabelOne
        </p>
        <ul
          className="emotion-7"
        >
          <li
            className="first"
          >
            <button
              aria-label="first page"
              className="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeLarge emotion-8"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              type="button"
            >
              <svg
                className="emotion-9"
              />
            </button>
          </li>
          <li>
            <button
              aria-label="previous page"
              className="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeLarge emotion-8"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              type="button"
            >
              <svg
                className="emotion-9"
              />
            </button>
          </li>
          <li
            className="MuiTypography-root MuiTypography-body2 selected emotion-12"
            onClick={[Function]}
          >
            1
          </li>
          <li
            className="MuiTypography-root MuiTypography-body2 emotion-12"
            onClick={[Function]}
          >
            2
          </li>
          <li>
            <button
              aria-label="next page"
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge emotion-8"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <svg />
            </button>
          </li>
          <li
            className="last"
          >
            <button
              aria-label="last page"
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge emotion-8"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <svg />
            </button>
          </li>
          <li
            className="emotion-16"
          >
            <div
              className="MuiFormControl-root emotion-17"
            >
              <div
                className="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl emotion-18"
                onClick={[Function]}
              >
                <div
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  className="MuiSelect-select MuiSelect-outlined MuiInputBase-input emotion-19"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onMouseDown={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  <p
                    className="MuiTypography-root MuiTypography-body2 emotion-20"
                  >
                    100
                  </p>
                </div>
                <input
                  aria-hidden={true}
                  aria-invalid={false}
                  className="MuiSelect-nativeInput emotion-21"
                  disabled={false}
                  onAnimationStart={[Function]}
                  onChange={[Function]}
                  required={false}
                  tabIndex={-1}
                  value={100}
                />
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined emotion-22"
                  color="#999999"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
            </div>
          </li>
        </ul>
        <ul
          className="emotion-23"
        >
          <li
            className="first"
          >
            <button
              aria-label="first page"
              className="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeLarge emotion-8"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              type="button"
            >
              <svg
                className="emotion-9"
              />
            </button>
          </li>
          <li>
            <button
              aria-label="previous page"
              className="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeLarge emotion-8"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              type="button"
            >
              <svg
                className="emotion-9"
              />
            </button>
          </li>
          <li
            className="MuiTypography-root MuiTypography-body2 selected emotion-12"
            onClick={[Function]}
          >
            1
          </li>
          <li
            className="MuiTypography-root MuiTypography-body2 emotion-12"
            onClick={[Function]}
          >
            2
          </li>
          <li>
            <button
              aria-label="next page"
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge emotion-8"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <svg />
            </button>
          </li>
          <li
            className="last"
          >
            <button
              aria-label="last page"
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge emotion-8"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <svg />
            </button>
          </li>
          <li
            className="emotion-16"
          >
            <div
              className="MuiFormControl-root emotion-17"
            >
              <div
                className="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl emotion-18"
                onClick={[Function]}
              >
                <div
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  className="MuiSelect-select MuiSelect-outlined MuiInputBase-input emotion-19"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onMouseDown={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  <p
                    className="MuiTypography-root MuiTypography-body2 emotion-20"
                  >
                    100
                  </p>
                </div>
                <input
                  aria-hidden={true}
                  aria-invalid={false}
                  className="MuiSelect-nativeInput emotion-21"
                  disabled={false}
                  onAnimationStart={[Function]}
                  onChange={[Function]}
                  required={false}
                  tabIndex={-1}
                  value={100}
                />
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined emotion-22"
                  color="#999999"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
